web/packages/teleterm/src/ui/Search/actions.tsx

```diff
@@ -43,13 +43,16 @@ export interface ParametrizedAction {
   searchResult: SearchResult;
   preventAutoClose?: boolean;
   parameter: {
-    getSuggestions(): Promise<string[]>;
+    getSuggestions(): Promise<Parameter[]>;
+    /** Disables providing new values. */
+    allowOnlySuggestions?: boolean;
     placeholder: string;
   };
-
-  perform(parameter: string): void;
+  perform(parameter: Parameter): void;
 }
 
+export type Parameter = { key: string; display: string };
```

Comment:
Let's add a JSDoc to the `display` field explaining that the input value provided by the user will be matched against this field.